### PR TITLE
[ENG-6185][ENG-6186][ENG-6187][ENG-6188] additional searchable metadata for institutional dashboard

### PR DESCRIPTION
**Pull request by aaxelb**
## Purpose
add searchable metadata that will be used by the new™️ institutional dashboard

## Changes
(recommend reading commit-by-commit)
- update metadata test helpers for more helpful feedback when adding metadata properties
- add `osf:hasOsfAddon` property (supplemental)
  - identify addons with `urn:osf.example:addons:...`
- add `osf:storageRegion` property (supplemental)
  - identify regions with `https://api.osf.example/v2/regions/.../`
- add `prov:qualifiedAttribution` (non-supplemental)
  - qualifies contributorship with a role: `osf:admin-contributor`, `osf:write-contributor`, or `osf:readonly-contributor` (and could add more nuanced/helpful contributor roles in the future)
- add `osf:storageByteCount` property (supplemental)
  - small refactor of `update_storage_usage_cache`:
    - add `compute_storage_usage_total` for the actual query
    - update query to use named params
    - replace a join with a separate `ContentType.objects.get_for_model()` call (which also allows for non-node file targets)
  - add `get_storage_usage_total` function (alternative to `Node.storage_usage` that waits for it to compute)

## QA Notes

Please make verification statements inspired by your code and what your code touches.
- Verify changes in the described properties are reflected in the search api (or the institutional dashboard) shortly enough
- Verify existing search indexing does not break

What are the areas of risk?

Any concerns/considerations/questions that development raised?

## Ticket

- `prov:qualifiedAttribution`: [ENG-6185](https://openscience.atlassian.net/browse/ENG-6185)
- `osf:storageRegion`: [ENG-6186](https://openscience.atlassian.net/browse/ENG-6186)
- `osf:storageByteCount`: [ENG-6187](https://openscience.atlassian.net/browse/ENG-6187)
- `osf:hasOsfAddon`: [ENG-6188](https://openscience.atlassian.net/browse/ENG-6188)

[ENG-6185]: https://openscience.atlassian.net/browse/ENG-6185?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: osf_tests/metadata/expected_metadata_files/project_monthly_supplement.turtle [hunk 1]
```diff
@@ -2,6 +2,7 @@
 @prefix dcterms: <http://purl.org/dc/terms/> .
 @prefix foaf: <http://xmlns.com/foaf/0.1/> .
 @prefix osf: <https://osf.io/vocab/2022/> .
+@prefix skos: <http://www.w3.org/2004/02/skos/core#> .
```
Comment (aaxelb) [inline osf_tests/metadata/expected_metadata_files/project_monthly_supplement.turtle:5]: remove?

File: osf/metadata/osf_gathering.py [hunk 2]
```diff
@@ -77,6 +77,16 @@ def pls_get_magic_metadata_basket(osf_item) -> gather.Basket:
...
+OSF_FILEVERSION = {
+    DCTERMS.created: None,
+    DCTERMS.creator: OSF_AGENT_REFERENCE,
+    DCTERMS.extent: None,
+    DCTERMS.modified: None,
+    DCTERMS.requires: None,
+    DCTERMS['format']: None,
+    OSF.versionNumber: None,
+}
+
```
Comment (aaxelb) [inline osf/metadata/osf_gathering.py:94]: unnecessary in the map -- consider removing

File: osf/metadata/osf_gathering.py [hunk 3]
```diff
@@ -688,6 +694,8 @@ def _gather_fileversion(fileversion, fileversion_iri):
     version_sha256 = (fileversion.metadata or {}).get('sha256')
     if version_sha256:
         yield (fileversion_iri, DCTERMS.requires, checksum_iri('sha-256', version_sha256))
+    if fileversion.region is not None:
+        yield (fileversion_iri, OSF.storageRegion, rdflib.URIRef(fileversion.region.absolute_api_v2_url))
```
Comment (aaxelb) [inline osf/metadata/osf_gathering.py:714]: prefLabel here too?

**Review by aaxelb — COMMENTED** [review]
(comments from group review)

**Comment by felliott** [comment]
LGTM, Abram found a few minor things that he will address, but no major blockers.

File: osf_tests/metadata/test_serialized_metadata.py [hunk 4]
```diff
@@ -260,6 +261,13 @@ def setUp(self):
                 doi_prefix='11.rp',
             ),
         )
+        self.reg_file = create_test_file(
+            self.registration,
+            self.user,
+            filename='my-reg-file.blarg',
+            size=17,
+            sha256='6ac3c336e4094835293a3fed8a4b5fedde1b5e2626d9838fed50693bba00af0e',
```
Comment (cslzchen) [inline osf_tests/metadata/test_serialized_metadata.py:269]: Do we actually need a valid SHA in test?

**Review by cslzchen — APPROVED** [review]
LGTM 🎆

**Pull request merged**